bridges/otellogrus/hook.go

```diff
@@ -0,0 +1,255 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package otellogrus provides a [Hook], a [logrus.Hook] implementation that
+// can be used to bridge between the [github.com/sirupsen/logrus] API and
+// [OpenTelemetry].
+//
+// # Record Conversion
+//
+// The [logrus.Entry] records are converted to OpenTelemetry [log.Record] in
+// the following way:
+//
+//   - Time is set as the Timestamp.
+//   - Message is set as the Body using a [log.StringValue].
+//   - Level is transformed and set as the Severity. The SeverityText is not
+//     set.
+//   - Fields are transformed and set as the attributes.
+//
+// The Level is transformed by using the static offset to the OpenTelemetry
+// Severity types. For example:
+//
+//   - [slog.LevelDebug] is transformed to [log.SeverityDebug]
+//   - [slog.LevelInfo] is transformed to [log.SeverityInfo]
+//   - [slog.LevelWarn] is transformed to [log.SeverityWarn]
+//   - [slog.LevelError] is transformed to [log.SeverityError]
+//
+// Attribute values are transformed based on type:
+//
+//   - booleans are transformed into [log.Bool]
+//   - byte arrays are transformed into [log.Bytes]
+//   - float64 are transformed into [log.Float64]
+//   - int are transformed into [log.Int]
+//   - int64 are transformed into [log.Int64]
+//   - string are transformed into [log.String]
+//
+// Any other type is transformed into a string.
+//
+// [OpenTelemetry]: https://opentelemetry.io/docs/concepts/signals/logs/
+package otellogrus // import "go.opentelemetry.io/contrib/bridges/otellogrus"
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/global"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+)
+
+const bridgeName = "go.opentelemetry.io/contrib/bridges/otellogrus"
+
+type config struct {
+	provider log.LoggerProvider
+	scope    instrumentation.Scope
+
+	levels []logrus.Level
+}
+
+func newConfig(options []Option) config {
```

Comment:
We miss testing of default configuration and when `WithInstrumentationScope` is passed.